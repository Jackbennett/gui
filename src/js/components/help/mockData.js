export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
