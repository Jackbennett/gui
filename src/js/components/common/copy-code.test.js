import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CopyCode from './copy-code';
import { undefineds } from '../../../../tests/mockData';

describe('CopyCode Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<CopyCode code="sudo it all!" />);
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as intended', async () => {
    const submitCheck = jest.fn();
    document.execCommand = jest.fn(() => true);
    const ui = <CopyCode code="sudo it all!" onCopy={submitCheck} withDescription={true} />;
    const { rerender } = render(ui);

    expect(screen.queryByText(/Copied to clipboard/i)).not.toBeInTheDocument();
    userEvent.click(screen.getByRole('button', { name: /Copy to clipboard/i }));
    expect(submitCheck).toHaveBeenCalledTimes(1);
    expect(document.execCommand).toHaveBeenCalledTimes(1);
    expect(screen.queryByText(/Copied to clipboard/i)).toBeInTheDocument();
    jest.advanceTimersByTime(6000);
    await waitFor(() => rerender(ui));
    expect(screen.queryByText(/Copied to clipboard/i)).not.toBeInTheDocument();
  });
});
