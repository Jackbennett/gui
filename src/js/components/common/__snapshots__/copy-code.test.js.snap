// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyCode Component renders correctly 1`] = `
<body>
  <div>
    <div
      class="code"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        style="float: right; margin: -20px 0px 0px 10px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <span
        style="white-space: pre-wrap; word-break: break-word;"
      >
        sudo it all!
      </span>
    </div>
    <p />
  </div>
</body>
`;
